### PR TITLE
Variants of UCUM in Positional/ Location Search #1100

### PR DESCRIPTION
**Pull request by prb112**
Signed-off-by: Paul Bastide <pbastide@us.ibm.com>

### CODE REVIEW

File: fhir-search/src/main/java/com/ibm/fhir/search/location/uom/standard/StatuteLengthUnits.java [hunk 1]
```diff
@@ -46,7 +50,11 @@
 
     MILE("mi_us", 8 * 40 * 16.5),
     MILE_CI("MI_US", 8 * 40 * 16.5),
-    
+    // Support for common variants that are not in UCUM
+    MILE_VARIANT_MIS("mis", 8 * 40 * 16.5),
+    MILE_VARIANT_MILE("mile", 8 * 40 * 16.5),
+    MILE_VARIANT_MILES("miles", 8 * 40 * 16.5),
```
Comment (lmsurpre) [inline fhir-search/src/main/java/com/ibm/fhir/search/location/uom/standard/StatuteLengthUnits.java:56]: no `mi` ?

File: fhir-search/src/main/java/com/ibm/fhir/search/location/uom/standard/StatuteLengthUnits.java [hunk 2]
```diff
@@ -16,6 +16,10 @@
 public enum StatuteLengthUnits {
     FOOT("ft_us", 1.0),
     FOOT_CI("FT_US", 1.0),
+    // Support for common variants that are not in UCUM
+    FOOT_VARIANT_FTS("fts", 1.0),
+    FOOT_VARIANT_FOOT("foot", 1.0),
+    FOOT_VARIANT_FEET("feet", 1.0),
```
Comment (lmsurpre) [inline fhir-search/src/main/java/com/ibm/fhir/search/location/uom/standard/StatuteLengthUnits.java:22]: not `ft`?

File: fhir-search/src/main/java/com/ibm/fhir/search/location/uom/standard/MetricUnits.java [hunk 3]
```diff
@@ -30,13 +30,21 @@
...
     HECTOMETRE("hm", 2.0),
     HECTOMETRE_CI("HM", 2.0),
     DECAMETRE("dam", 1.0),
     DECAMETRE_CM("DAM", 1.0),
     METRE("m", 0.0),
     METRE_CI("M", 0.0),
+    // The following 'variants' are not standard UCUM, but helpful for various common usage
+    METRE_VARIANT_MS("ms", 0.0),
+    METRE_VARIANT_METER("meter", 0.0),
+    METRE_VARIANT_METERS("meters", 0.0),
```
Comment (lmsurpre) [inline fhir-search/src/main/java/com/ibm/fhir/search/location/uom/standard/MetricUnits.java:47]: no `m`?

File: fhir-search/src/main/java/com/ibm/fhir/search/location/uom/standard/StatuteLengthUnits.java (hunk 2 above)
Comment (prb112) [reply fhir-search/src/main/java/com/ibm/fhir/search/location/uom/standard/StatuteLengthUnits.java:22]: thanks, it's hard to see the tree in the forest

File: fhir-search/src/main/java/com/ibm/fhir/search/location/uom/standard/StatuteLengthUnits.java (hunk 1 above)
Comment (prb112) [reply fhir-search/src/main/java/com/ibm/fhir/search/location/uom/standard/StatuteLengthUnits.java:56]: thanks, it's hard to see the tree in the forest

**Comment by prb112** [comment]
Added ft/mi in latest commit

**Comment by prb112** [comment]
METRE("m", 0.0),
    METRE_CI("M", 0.0),

It's actually already covered :) it's part of UCUM

File: fhir-search/src/main/java/com/ibm/fhir/search/location/uom/UOMManager.java [hunk 4]
```diff
@@ -186,6 +186,7 @@ public static final Double getUnitToMetersFactor(final String unit) {
         // Supports UCUM with [mi_us] or mi_us
         if(factor == null && unit != null && 
                 unit.startsWith("[") && unit.endsWith("]")) {
+            System.out.println(unit.substring(1,unit.length() -1));
```
Comment (lmsurpre) [inline fhir-search/src/main/java/com/ibm/fhir/search/location/uom/UOMManager.java:189]: remove?

Comment (prb112) [reply fhir-search/src/main/java/com/ibm/fhir/search/location/uom/UOMManager.java]: Yes - Fixed in latest commit

**Pull request merged**